app/models/miq_server.rb

```diff
@@ -627,10 +627,12 @@ def messaging_health_check
     broker = MiqQueue.messaging_client("health_check")
     return if broker.nil?
 
-    broker.publish_message(:service => "manageiq.liveness-check", :message => "test message", :payload => {})
-    broker.subscribe_messages(:service => "manageiq.liveness-check") { break }
+    # Fail health check if list of topics can't be retrieved
+    broker.topics
   rescue => err
     _log.error("Messaging health check failed: #{err}")
     shutdown_and_exit(1)
+  ensure
+    broker.close
```

Comment:
You need to handle the case where broker is nil here  because ensure will still trigger on the return up here https://github.com/ManageIQ/manageiq/pull/23020/files#diff-0c71d22135646ed908fa6a66bec47aa6024935e7d9ed66e4a99c64f9355b3307R628

The other option is to have an explicit begin/rescue/ensure like

```
def messaging_health_check
  broker =
  return if broker.nil?

  begin
    ...
  rescue => err
    ...
  ensure
    broker.close
  end
end